[16.0][mig] product_multi_image

IJOL opened the pull request.

pedrobaeza committed:
product_multi_image: Module totally redone from old product_images

yajo committed:
[IMP] product_multi_image: Split model in abstract & concrete + better icon

yajo committed:
Fix bugs found in review time, improve README.

pedrobaeza committed:
product_multi_image: Handle images in variants

lasley committed:
[MIG] product_multi_image: Upgrade to v9
* Upgrade product_multi_image to v9
* Add server-tools to oca_dependencies

lasley committed:
[MIG] product_multi_image: Upgrade to v10
* Migrate product_multi_image to v10
* Remove old api from `product.product`
* Add test coverage for hooks and variant count
* Fix oca depends
* Add uninstall hook

simahawk committed:
[fix] prod image update broken w/ no image on template #228

Commit:
[IMP] product_multi_image: Improve uninstall hook to move images from
multi to single mode

Commit:
[FIX] product_multi_image: Do not resize large image to small/medium
when saved from single image

Commit:
[FIX] product_multi_image: Fix image removal from variants

MPoch-PlanetaTIC committed:
[MIG] product_multi_image: Migration to 12.0

rousseldenis committed:
[14.0][ADD] product_category_code_unique

rousseldenis committed:
[14.0][IMP] product_category_code_unique: Add sequence

hailangvn committed:
[IMP] product_multi_image: black, isort, prettier

hailangvn committed:
[MIG] product_multi_image: Migration to 14.0

hailangvn committed:
[IMP] product_multi_image: show images in product variant easy edit view

ivantodorovich committed:
[ADD] purchase_product_template_tags

ivantodorovich committed:
[ADD] stock_product_template_tags

oca-travis committed:
[UPD] Update purchase_product_template_tags.pot

oca-travis committed:
[UPD] Update stock_product_template_tags.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
purchase_product_template_tags 14.0.1.1.0

OCA-git-bot committed:
stock_product_template_tags 14.0.1.1.0

sebalix committed:
[ADD] New module 'product_main_supplierinfo'

oca-travis committed:
[UPD] Update product_main_supplierinfo.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

Kev-Roche committed:
Added translation using Weblate (French)

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-attribute-14.0/product-attribute-14.0-purchase_product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-purchase_product_template_tags/it/

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-attribute-14.0/product-attribute-14.0-stock_product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-stock_product_template_tags/it/

rousseldenis committed:
[14.0][IMP] product_category_code_unique: Add False company_id data

oca-travis committed:
[UPD] Update product_category_code_unique.pot

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

Kev-Roche committed:
[FIX] product_stock_state product_template_view

flachica committed:
[MIG] product_multi_image 15.0

TDu committed:
product_main_supplierinfo: fix default sellers for variant

If no specific record are found by the first filter. The Odoo default
method will return all active supplier info.
Which will include also supplier info record related to other specific
variant.

With this change if some supplier info related to the product
variant exist, they will be returned instead.

Also add a sort on the price to have the same behaviour than on a
purchase order line for the price selected.

OCA-git-bot committed:
product_main_supplierinfo 14.0.1.0.1

rolandojduartem committed:
[MIG] product_category_code_unique: Migrate to v15.0.

Commit:
[UPD] Update product_category_code_unique.pot

OCA-git-bot committed:
[UPD] README.rst

OKT96 committed:
[MIG] purchase_product_template_tags: Migration to 15.0

MarcBForgeFlow committed:
[ADD] product_category_description

dhx committed:
[MIG] stock_product_template_tags: Migration to 15.0

Commit:
[UPD] Update product_multi_image.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update purchase_product_template_tags.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update stock_product_template_tags.pot

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-attribute-15.0/product-attribute-15.0-purchase_product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-purchase_product_template_tags/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-attribute-15.0/product-attribute-15.0-stock_product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-stock_product_template_tags/it/

klodr committed:
[MIG] product_category_description: Migration to 14.0

dsolanki-initos committed:
[IMP] purchase_product_template_tags: pre-commit stuff

dsolanki-initos committed:
[MIG] purchase_product_template_tags: Migration to 16.0

dsolanki-initos committed:
[IMP] stock_product_template_tags: pre-commit stuff

dsolanki-initos committed:
[MIG] stock_product_template_tags: Migration to 16.0

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-15.0/product-attribute-15.0-product_multi_image
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_multi_image/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (14 of 14 strings)

Translation: product-attribute-15.0/product-attribute-15.0-product_multi_image
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_multi_image/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-15.0/product-attribute-15.0-product_category_code_unique
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_category_code_unique/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: product-attribute-15.0/product-attribute-15.0-product_multi_image
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_multi_image/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_main_supplierinfo/it/

Commit:
[UPD] Update product_category_description.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_category_description
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_category_description/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_category_description
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_category_description/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_main_supplierinfo/es/

TDu committed:
Fix typo in product_main_supplierinfo

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[ADD] product_attachment_zipped_download: New module

TT48578

[UPD] Update product_attachment_zipped_download.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-15.0/product-attribute-15.0-product_attachment_zipped_download
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_attachment_zipped_download/

legalsylvain committed:
[ADD] product_usability: New module that adds a new main entry menu for models present
in 'product' module. (product.product, product.category, etc...)

chienandalu committed:
[ADD] product_supplierinfo_import_by_barcode: New module

TT49604

IJOL committed:
[MIG] product_multi_image: Migration to 16.0

IJOL commented:
Waiting for a merge in server-tools

thomaspaulb commented:
@IJOL Can you rebase to see what this now does

IJOL commented:
Green Green, needs review thought

AurelijaNorvaisaite commented:
@jacekmichalski @rousseldenis @thomaspaulb maybe you could review the code?

AurelijaNorvaisaite commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -0,0 +1,234 @@
+                        {
+                            # "storage": "db",
+                            "name": "Image 1",
+                            # "file_db_store": self.transparent_image,
```
I think these comments should be removed

bosd commented on product_multi_image/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+    "license": "AGPL-3",
+    "website": "https://github.com/OCA/product-attribute",
+    "category": "Sales Management",
+    "pre_init_hook": "post_init_hook",
```
The pre ulinit hook, is triggering a post init hook? A bit confusing.. Is that correct?

IJOL replied:
Thanks for catching this one, this slipped from previous version and culprit of some headaches

bosd reviewed (changes requested):
Thanks for your work on this. Please remove the new introduced comments in the migration commit.

IJOL commented:
> Thanks for your work on this. Please remove the new introduced comments in the migration commit.

I dont think I fully understand this remark..

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -92,7 +90,14 @@ def test_restrict_one_image(self):
+                0,
+                0,
+                {
+                    # "storage": "db",
```
Please remove this line with the comment

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -132,9 +137,10 @@ def test_create_variant_afterwards(self):
                         0,
                         {
-                            "storage": "db",
+                            # "storage": "db",
```
Please remove this line with the comment

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -132,9 +137,10 @@ def test_create_variant_afterwards(self):
+                            # "storage": "db",
                             "name": "Image 1",
-                            "file_db_store": self.transparent_image,
+                            # "file_db_store": self.transparent_image,
```
Please remove this line with the comment

bosd reviewed (changes requested):
To clarify this is what I was pointing at.

yvaucher commented:
/ocabot migration product_multi_image

yvaucher reviewed (commented):
LGTM

Code review

bealdav committed:
[ADD] product_english_name: module which displays the product name in English near the products name in user's language

OCA-git-bot committed:
Merge PR #1576 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update product_pricelist_alternative.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_alternative
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_alternative/

psugne commented:
@bosd @yvaucher How is going here? Any timeline when changes will be merged to 16.0 version?

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -0,0 +1,225 @@
+                {"image_1920": self.grey_image},
+            )
+        ]
+        self.product_template.refresh()
```
```suggestion
        self.product_template.invalidate_cache()
```

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -0,0 +1,225 @@
+    def test_edit_image_variant(self):
+        text = "Test name changed"
+        self.product_1.image_ids[0].name = text
+        self.product_template.refresh()
```
```suggestion
        self.product_template.invalidate_cache()
```

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -0,0 +1,225 @@
+    def test_remove_image_all_variants(self):
+        self.product_1.image_ids = [(3, self.product_1.image_ids[0].id)]
+        self.product_2.image_ids = [(3, self.product_2.image_ids[0].id)]
+        self.product_template.refresh()
```
```suggestion
        self.product_template.invalidate_cache()
```

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -0,0 +1,225 @@
+
+    def test_remove_image_variant(self):
+        self.product_1.image_ids = [(3, self.product_1.image_ids[0].id)]
+        self.product_template.refresh()
```
```suggestion
        self.product_template.invalidate_cache()
```

bosd commented on product_multi_image/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,15 @@
+* When you change the image on the product variant, the preview image of the
+  *Images* tab doesn't get refreshed until you refresh the browser, or if you
+  go to its template, but the image has been actually saved!
```
```suggestion
```

This actually has been fixed. So can be removed from the roadmap.

bosd commented on product_multi_image/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,15 @@
+* Add logic for handling to add the same image in several variants to a
+  already in another variant for not duplicating bytes.
+* Provide proper migration scripts from module product_images from 7.0.
+* Migrate to v8 api when https://github.com/odoo/odoo/issues/10799 gets fixed.
```
```suggestion
```

Quite old comment, this can be removed.

bosd commented on product_multi_image/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,15 @@
+* When you change the image on the product variant, the preview image of the
+  *Images* tab doesn't get refreshed until you refresh the browser, or if you
+  go to its template, but the image has been actually saved!
+* The field "Available in these variants" appears when opening the image
```
I have found a "similar" issue to still be present. Maybe remove the field from the view when coming form a product variant. Or at least fix the domain it is searching on.

bosd commented on product_multi_image/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,15 @@
+  go to its template, but the image has been actually saved!
+* The field "Available in these variants" appears when opening the image
+  from the product variant.
+* Add logic for handling to add images with the same name that another variant
```
afaik, this is all outdated. Can be removed.

bosd commented on product_multi_image/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+    "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "website": "https://github.com/OCA/product-attribute",
+    "category": "Sales Management",
```
```suggestion
    "category": "Product",
    "development_status": "Beta",
    "summary": "Add multiple images for a product, a.k.a. an image gallery.",
```

bosd commented on product_multi_image/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+        "images/db.png",
+        "images/file.png",
+        "images/url.png",
+    ],
```
```suggestion
```
This tage has been deprecated since V15. 
The image files it is pointing to are rather old. Version 8 style! so better to remove them.
Or replace and improve documentation if your'e up for that.

bosd commented on product_multi_image/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,15 @@
+  already in another variant for not duplicating bytes.
+* Provide proper migration scripts from module product_images from 7.0.
+* Migrate to v8 api when https://github.com/odoo/odoo/issues/10799 gets fixed.
+* If you try to sort images before saving the product variant or template, you
+  will get an error similar to ``DataError: invalid input syntax for integer:
+  "one2many_v_id_62"``. This bug has not been fixed yet, but a workaround is to
+  save and edit again to sort images.
```
This whole section is no longer relevant. So better remove.
```suggestion
```

bosd commented on product_multi_image/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,26 @@
+#. Go to *Sales > Products > Products* and choose a product template.
+#. Go to the *Images* tab.
+#. Add a new image or edit the existing ones.
+#. You can select for which variants you want to make available the image.
```
```suggestion
#. You can select for which variants you want to make the image available.
```

bosd commented on product_multi_image/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,26 @@
+#. Add a new image or edit the existing ones.
+#. You can select for which variants you want to make available the image.
+   Keep it empty for making visible in all.
+#. Refresh the page.
```
```suggestion
```
No longer needed.

bosd commented on product_multi_image/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,26 @@
+#. If you add an image here, the image is actually added to the product
+   template, and restricted to this variant.
+#. When editing an existing image, the image is changed generally for all
+   the variants where is enabled, not only for this variant.
```
```suggestion
   the variants where it is enabled, not only for this variant.
```

bosd commented on product_multi_image/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,26 @@
+   the variants where is enabled, not only for this variant.
+#. When removing an image from this form, if the image is only in this variant,
+   the image is removed. Otherwise, the image gets restricted to the rest of
+   the variants where is available.
```
```suggestion
   the variants where it is used.
```

bosd commented on product_multi_image/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,26 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/135/9.0
```
```suggestion
```
Runbot 9 is no longer there. Better remove this section.

bosd commented on product_multi_image/tests/test_product_multi_image.py:
```diff
@@ -0,0 +1,225 @@
+            image.product_variant_count,
+            1,
+        )
+
```
Maybe add an test to import image from url. As it is failling with an error currently.

Aldeigja replied:
> Maybe add an test to import image from url. As it is failling with an error currently.

@bosd I think this issue has been fixed in this PR - https://github.com/OCA/server-tools/pull/3168

bosd commented on product_multi_image/models/product_product.py:
```diff
@@ -0,0 +1,86 @@
+        inverse="_inverse_image_ids",
+    )
+
+    # image, image_medium, image_small fields are not available since 13.0
```
```suggestion
```

bosd reviewed (changes requested):
Thanks for all your work on this module. :+1: 
I have found some warnings in the log to call deprecated flush method.Added code suggestions to fix it.
I noticed there is some leftover from previous migration.


I have found some things that didn't work during functional tests.

### **On product variants incorrect values are shown in the variants section.**
_To reproduce...._
Add an image from the product variant.Go to a product variant, 
On Images tab click ADD.
In the screen that opens there is a "Visible in these Variants" section.
When clicking on the dropdown. 
You can select another totally unrelated product (variant). 
Which is actually quite confusing. 
It should be restricted automatically to the variant you are editing. 
But when you click save. 
It actually does restrict the image to the variant you are editing.


### **First image is not saved when creating a new product.**
_To reproduce...._
Create a new product.
Add one image, click save and new,
Add the second product image.
Hit the save button (The cloud symbol) Then you will notice the content of the first image is deleted.

### **Image is not saved on a newly created product**
_To reproduce...._
Create a new product.
Go to images tab. Click add. select a image.Click "Save & Close".
You will see a preview of the image on the images tab.
Once you click the save button (The cloud symbol). 
The image preview is deleted.


### **Image is deleted when changing the sequence.**
_To reproduce...._
Create or select an existing product.
click on save button (The cloud icon).
Click the add button to add a image.
Upload an image click save and close.
Back in the product click on save button (The cloud icon).
Upload another image click save and close.
Back in the product click on save button (The cloud icon).
Click on the image you last uploaded to edit it.
Change sequence from 10 to a lower number.. e.g. 9
Click save
Now it is ok. Earlier my test was failling. Likely when not pressing the cloud icon.

### **It is possible to add an empty image**
_Rather more a "feature then a big bug"_
_To reproduce...._
Create or select an existing product.
On Images tab click ADD.
Fill nothing in... Leave all empty. Click on save button.You have just added an empty picture to your product.

### **Adding an image from url results in an error.**
Use the url field when adding an image.
<details>
```
Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1651, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1678, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 1882, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/addons/website/models/ir_http.py", line 237, in _dispatch
    response = super()._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 734, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 42, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 468, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 453, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/odoo/models.py", line 6614, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File "/opt/odoo/odoo/models.py", line 6325, in _onchange_eval
    method_res = method(self)
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/base_multi_image/models/image.py", line 91, in _onchange_load_from
    self.name, self.extension = os.path.splitext(filename)
AttributeError: 'base_multi_image.image' object has no attribute 'extension'

The above server error caused the following client error:
null
```
</details>

These 2 commits should be squashed:
![image](https://github.com/user-attachments/assets/4cfd9984-61dd-4dae-bdaa-c0975dde8dda)

OCA-git-bot committed:
Merge PR #1680 into 16.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_alternative
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_alternative/it/

OCA-git-bot committed:
Merge PR #1732 into 16.0

Signed-off-by HaraldPanten

Commit:
[UPD] Update product_english_name.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1515 into 16.0

Signed-off-by rousseldenis

OCA-git-bot committed:
[BOT] post-merge updates

IJOL committed:
Update product_multi_image/models/product_product.py

Co-authored-by: bosd <c5e2fd43-d292-4c90-9d1f-74ff3436329a@anonaddy.me>

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_english_name
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_english_name/it/

SandieFavre committed:
Added translation using Weblate (French)

SandieFavre committed:
Translated using Weblate (French)

Currently translated at 59.0% (13 of 22 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_simulation
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_simulation/fr/

SandieFavre committed:
Added translation using Weblate (French)

SandieFavre committed:
Translated using Weblate (French)

Currently translated at 71.4% (5 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_simulation_margin
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_simulation_margin/fr/

legalsylvain committed:
[DOC] product_usability: add a note for V18 migration

legalsylvain committed:
[FIX] uom_category_active: CI is broken. because onchange doesn't raise an error. Swith from an onchange logic to an create / write logic to be sure to handle all the cases

OCA-git-bot committed:
Merge PR #1760 into 16.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1757 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update product_stock_state.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_stock_state
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_stock_state/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 28.5% (8 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/fr/

samibc2c committed:
Translated using Weblate (French)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_dimension/fr/

OCA-git-bot committed:
Merge PR #1657 into 16.0

Signed-off-by legalsylvain

Commit:
[UPD] Update product_usability.pot

OCA-git-bot committed:
[BOT] post-merge updates

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 93.1% (27 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 71.4% (5 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_container_deposit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_container_deposit/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_total_weight_from_packaging
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_total_weight_from_packaging/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 90.0% (9 of 10 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_level_purchasable
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_level_purchasable/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 14.7% (5 of 34 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_level
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_level/fr/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 100.0% (22 of 22 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_dimension/fr/

chaule97 committed:
[FIX]: Use cls.addClassCleanup(cls.loader.restore_registry)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_usability
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_usability/it/

baguenth committed:
[FIX] product_uom_measure_type: Fixed the issue, that the module cannot be installed on migrated databases which already have a column 'measure_type'

OCA-git-bot committed:
Merge PR #1779 into 16.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

mmequignon committed:
[IMP] product_main_supplierinfo: pre-commit stuff

mmequignon committed:
[MIG] product_main_supplierinfo: Migration to 16.0

OCA-git-bot committed:
Merge PR #1485 into 16.0

Signed-off-by dreispt

Commit:
[UPD] Update product_internal_reference_generator.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1763 into 16.0

Signed-off-by dreispt

OCA-git-bot committed:
Merge PR #1786 into 16.0

Signed-off-by dreispt

Commit:
[UPD] Update product_main_supplierinfo.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_internal_reference_generator
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_internal_reference_generator/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_main_supplierinfo/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_main_supplierinfo/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_internal_reference_generator
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_internal_reference_generator/it/

OCA-git-bot committed:
Merge PR #1667 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update product_supplierinfo_import_by_barcode.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_supplierinfo_import_by_barcode
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_supplierinfo_import_by_barcode/

mymage committed:
Added translation using Weblate (Italian)

victoralmau committed:
[IMP] product_attachment_zipped_download: pre-commit stuff

victoralmau committed:
[MIG] product_attachment_zipped_download: Migration to 16.0

TT52271

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (38 of 38 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_supplierinfo_import_by_barcode
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_supplierinfo_import_by_barcode/it/

manuelregidor committed:
[IMP] product_category_code_unique: black, isort, prettier

ahmedtrigui committed:
Added translation using Weblate (French)

ahmedtrigui committed:
Translated using Weblate (French)

Currently translated at 100.0% (4 of 4 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_category_active
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_category_active/fr/

legalsylvain committed:
[FIX] product_category_active: harmonization of category display.
In odoo, archived items are not displayed by default. (see users view, products, etc...)
The product_category_active module doesn't follow this convention because it is changing the context of the action of product.category. As a result, all categories are displayed, polluting visibility with obsolete items.
- Remove the active_test in the context of the action.
- create a migration script to remove the active_test on existing instances.
- add filters to display archived and 'archived or active' categories, in case you want to see all categories

manuelregidor committed:
[MIG] product_category_code_unique: migration to 16

manuelregidor committed:
[IMP] product_category_code_unique: restriction options

Tisho99 committed:
[IMP] product_category_description: pre-commit stuff

Tisho99 committed:
[MIG] product_category_description: Migration to 16.0

OCA-git-bot committed:
Merge PR #1842 into 16.0

Signed-off-by HaraldPanten

Commit:
[UPD] Update product_category_description.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_category_description
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_category_description/

OCA-git-bot committed:
Merge PR #1834 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update product_category_active.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_category_active
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_category_active/

samibc2c committed:
Translated using Weblate (French)

Currently translated at 82.1% (23 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/fr/

samibc2c committed:
Translated using Weblate (French)

Currently translated at 50.0% (17 of 34 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_packaging_level
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_packaging_level/fr/

OCA-git-bot committed:
Merge PR #1822 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update product_attachment_zipped_download.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_attachment_zipped_download
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_attachment_zipped_download/

baguenth committed:
[FIX] product_packaging_level: Removed usage of undefined function

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_category_active
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_category_active/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_attachment_zipped_download
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_attachment_zipped_download/it/

OCA-git-bot committed:
Merge PR #1846 into 16.0

Signed-off-by HaraldPanten

OCA-git-bot committed:
[BOT] post-merge updates

AurelijaNorvaisaite commented:
@IJOL How is going here?

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_fixed_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_fixed_currency_rate/it/

OCA-git-bot committed:
[IMP] update dotfiles

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 93.1% (27 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/nl/

OCA-git-bot committed:
Merge PR #1870 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update stock_product_template_tags.pot

OCA-git-bot committed:
[BOT] post-merge updates

legalsylvain committed:
[FIX] product_print_category: Do no set current company by default on category model.
Rational: The categories are mainly created by xml code, because it requires to define qweb report. In that case, setting by default the company of the admin user doesn't makes sense. Also if categories are created by UI, in a multi-company context, you may or may not want the category to be global.

OCA-git-bot committed:
Merge PR #1869 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update purchase_product_template_tags.pot

OCA-git-bot committed:
[BOT] post-merge updates

carlosdauden committed:
[IMP] product_pricelist_supplierinfo: Add ignore_supplierinfo_margin option in pricelist item
TT55043

santostelmo committed:
[FIX] product_assortment ir_filters create vals_list

OCA-git-bot committed:
Merge PR #1885 into 16.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Add decimal precision to price fields

With this change we are sure that the prices shown on the report has the correct precision.

OCA-git-bot committed:
Merge PR #1887 into 16.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1884 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update product_pricelist_supplierinfo.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_supplierinfo/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_supplierinfo/it/

OCA-git-bot committed:
Merge PR #1881 into 16.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

carlosdauden committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_supplierinfo/es/

Shide committed:
[FIX] product_sticker: Allow to select which models can use Stickers

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_multi_category/tr/

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_manufacturer
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_manufacturer/tr/

OCA-git-bot committed:
Merge PR #1893 into 16.0

Signed-off-by rafaelbn

Commit:
[UPD] Update product_sticker.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_sticker
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_sticker/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_sticker
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_sticker/it/

milleniumkid committed:
Added translation using Weblate (Turkish)

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_dimension/tr/

milleniumkid committed:
Added translation using Weblate (Turkish)

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_main_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_main_supplierinfo/tr/

milleniumkid committed:
Added translation using Weblate (Turkish)

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_sale_description
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_sale_description/tr/

uvid-gordana committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_category_active
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_category_active/hr/

milleniumkid committed:
Added translation using Weblate (Turkish)

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 3.5% (1 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/tr/

milleniumkid committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (28 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/tr/

uvid-gordana committed:
Added translation using Weblate (Croatian)

davidbeckercbl committed:
Translated using Weblate (German)

Currently translated at 100.0% (23 of 23 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_template_tags
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_template_tags/de/

davidbeckercbl committed:
Translated using Weblate (German)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_code_unique
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_code_unique/de/

davidbeckercbl committed:
Translated using Weblate (German)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_manufacturer
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_manufacturer/de/

davidbeckercbl committed:
Translated using Weblate (German)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_dimension/de/

davidbeckercbl committed:
Translated using Weblate (German)

Currently translated at 82.1% (23 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/de/

Shide committed:
[IMP] product_sticker: Allow to choose size

OCA-git-bot committed:
Merge PR #1902 into 16.0

Signed-off-by Shide

Commit:
[UPD] Update product_sticker.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_sticker
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_sticker/

OCA-git-bot committed:
Merge PR #1829 into 16.0

Signed-off-by HaraldPanten

Commit:
[UPD] Update product_category_code_unique.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (52 of 52 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_sticker
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_sticker/it/

bosd commented:
ping @IJOL

IJOL committed:
Apply suggestions from code review

Co-authored-by: bosd <11499387+bosd@users.noreply.github.com>

IJOL committed:
[UPD] optimize image assignment logic in product template

IJOL committed:
[UPD] refine product image handling and update documentation for clarity

IJOL committed:
[UPD] improve documentation for image variant selection and behavior

bosd reviewed (approved):
Func test :+1:

yvaucher commented:
/ocabot merge nobump
